consensus/state.go

```diff
@@ -528,27 +538,38 @@ func (cs *ConsensusState) EnterPropose(height int, round int) {
 	cs.mtx.Lock()
 	defer cs.mtx.Unlock()
 	if cs.Height != height || round < cs.Round || (cs.Round == round && RoundStepPropose <= cs.Step) {
-		log.Info(Fmt("EnterPropose(%v/%v): Invalid args. Current step: %v/%v/%v", height, round, cs.Height, cs.Round, cs.Step))
+		log.Debug(Fmt("EnterPropose(%v/%v): Invalid args. Current step: %v/%v/%v", height, round, cs.Height, cs.Round, cs.Step))
 		return
 	}
 	log.Info(Fmt("EnterPropose(%v/%v). Current: %v/%v/%v", height, round, cs.Height, cs.Round, cs.Step))
 
+	var enterPrevote = make(chan struct{}, 1)
+
 	defer func() {
 		// Done EnterPropose:
 		cs.Round = round
 		cs.Step = RoundStepPropose
-		cs.newStepCh <- cs.getRoundState()
-
-		// If we already have the proposal + POL, then goto Prevote
-		if cs.isProposalComplete() {
-			go cs.EnterPrevote(height, cs.Round)
-		}
+		enterPrevote <- struct{}{}
```

Comment:
so part of the reason for doing this was to avoid going into prevote a second time with invalid args. but since we basically rely on that kind of thing everywhere maybe this construction is overly complicated and not really worth it